### ISSUE TITLE
Surveyor does not preserve question order

### ISSUE THREAD

**Issue by rsutphin** [NUARIG/surveyor#246]
While surveyor's question model has a `display_order` attribute, the DSL interpreter does not appear to ensure that this attribute is set to reflect the actual order of the questions in the survey definition — `display_order` is always zero. This means that surveyor presents questions in the order they are retrieved from the database, which is arbitrary.

This behavior was observed under Rails 3.1.1.

A workaround for this issue is to override the question model and change the default scope ordering to consider the surrogate key:

``` ruby
class Question < ActiveRecord::Base
  include Surveyor::Models::QuestionMethods

  default_scope :order => "display_order ASC, id ASC"

end
```

This will display in the questions in their insertion order, which is usually good enough. However, surveyor ought to explicitly preserve the ordering from the survey definition.

**Comment by yoon** [comment]
closed by 0e19f7de9b7f45b5bc934a0a1a87122511167831

**Comment by yoon** [comment]
6c9908bab4110e0d2a9637b620c374861be7265e
5fe9712da73bc1a81e4f27d9d4c49fb9ae98ff8b
